Client/TabManagement/Tab.swift

```diff
@@ -845,6 +841,19 @@ class Tab: NSObject {
 
         return .none
     }
+
+    // MARK: - ThemeApplicable
+
+    func applyTheme(theme: Theme) {
+        UITextField.appearance().keyboardAppearance = isPrivate ? .dark : (theme.type == .dark) ? .dark : .light
+    }
+
+    // MARK: - LegacyTheme
+
+    // This function needs to be in place since the LegacyTabManager still exists
+    func applyTheme() {
+        UITextField.appearance().keyboardAppearance = isPrivate ? .dark : (LegacyThemeManager.instance.currentName == .dark ? .dark : .light)
+    }
```

Comment:
I think we could remove it by removing the call to `tab.applyTheme()` in the TabManager. This means we need to add a new call to properly theme the tab when the tab is selected. I think this could be done in BrowserViewController under `didSelectedTabChange` delegate call of `TabManagerDelegate`. In BVC we have access to the new theme manager so we can pass in the current theme.

Reply:
ok perfect, i'm going to do it